types/tx.go

```diff
@@ -188,34 +176,15 @@ func (t TxRecordSet) Validate(maxSizeBytes int64, otxs Txs) error {
 		}
 	}
 
-	// create copies of each of the action-specific indexes so that order of the original
-	// indexes can be preserved.
-	addedCopy := sortedCopy(t.added)
-	removedCopy := sortedCopy(t.removed)
-	unmodifiedCopy := sortedCopy(t.unmodified)
-
+	// Check total size of included transactions
 	var size int64
-	for _, cur := range append(unmodifiedCopy, addedCopy...) {
+	for _, cur := range append(t.unmodified, t.added...) {
```

Comment:
this isn't your fault, but I'd do this as: 
```go
for _, txSlice := range []Txs{t.unmodified, t.added} {
   for _, cur := range txSlice {
      // current loop body
   }
}
```
which is what the current thing is doing: _except_ the current thing actually has some memory use side effects that you don't expect. (e.g. it makes the underlying array of the first slice (which is a view) bigger and adds the second elements to it, which are still hidden from the original view (so it's fine) but that hangs around till the `TxRecordSet` is gc'd, which... isn't great. And you don't need to allocate any memory here anyway because you have it all already.